Week1/homework/app.js

```diff
@@ -1,11 +1,179 @@
+    listElem.innerText = listText;
+    ul.appendChild(listElem);
+  }
+}
+
+function main() {
+  const root = document.getElementById('root');
+  const ul = document.createElement('ul');
+  root.appendChild(ul);
+  createUl(ul);
+}
+
+window.addEventListener('load', main);
+*/
+
+// Question 1.4
+
+const bookSpecified = {
+  booknumber0: {
+    title: 'title0',
+    language: 'language0',
+    author: 'author0',
+  },
+  booknumber1: {
+    title: 'title1',
+    language: 'language1',
+    author: 'author1',
+  },
+  booknumber2: {
+    title: 'title2',
+    language: 'language2',
+    author: 'author2',
+  },
+  booknumber3: {
+    title: 'title3',
+    language: 'language3',
+    author: 'author3',
+  },
+  booknumber4: {
+    title: 'title4',
+    language: 'language4',
+    author: 'author4',
+  },
+  booknumber5: {
+    title: 'title5',
+    language: 'language5',
+    author: 'author5',
+  },
+  booknumber6: {
+    title: 'title6',
+    language: 'language6',
+    author: 'author6',
+  },
+  booknumber7: {
+    title: 'title7',
+    language: 'language7',
+    author: 'author7',
+  },
+  booknumber8: {
+    title: 'title8',
+    language: 'language8',
+    author: 'author8',
+  },
+  booknumber9: {
+    title: 'title9',
+    language: 'language9',
+    author: 'author9',
+  },
+};
+
+// Question 1.5
+
+/*
+function createUl(ul) {
+  for (let key in bookSpecified) {
+    const titleText = bookSpecified[key].title;
+    const languageText = bookSpecified[key].language;
+    const authorText = bookSpecified[key].author;
+    const titleElem = document.createElement('h1');
+    const languageElem = document.createElement('p');
+    const authorElem = document.createElement('p');
+    const li = document.createElement('li');
+    titleElem.innerText = titleText;
+    languageElem.innerText = languageText;
+    authorElem.innerText = authorText;
+    li.appendChild(titleElem);
+    li.appendChild(languageElem);
+    li.appendChild(authorElem);
+    ul.appendChild(li);
+  }
+}
+
+function main() {
+  const root = document.getElementById('root');
+  const ul = document.createElement('ul');
+  root.appendChild(ul);
+  createUl(ul);
+}
+
+window.addEventListener('load', main);
+*/
+
+// Question 1.7
+
+const images = {
+  booknumber0: './img/1.jpg',
+  booknumber1: './img/1.jpg',
+  booknumber2: './img/2.jpg',
+  booknumber3: './img/3.jpg',
+  booknumber4: './img/4.jpg',
+  booknumber5: './img/5.jpg',
+  booknumber6: './img/6.jpg',
+  booknumber7: './img/7.jpg',
+  booknumber8: './img/8.jpg',
+  booknumber9: './img/9.jpg',
+};
+
+// Question 1.8
+
+function createUl(ul) {
+  for (const key of Object.keys(bookSpecified)) {
+    const titleText = bookSpecified[key].title;
+    const languageText = bookSpecified[key].language;
+    const authorText = bookSpecified[key].author;
+    const titleElem = document.createElement('h1');
+    const imageElem = document.createElement('img');
+    imageElem.setAttribute('src', `${images.key}`);
```

Comment:
You are searching for the `.key` but you don't want the actual string key but the value of the key, in that case you can use the bracket notion `images[key]`. The bracket notation can contain variables. 

Also, since it's already a string there is no need to wrap it in quotes; `'${images.key}'` can be written as `images[key]`.

Reply:
Got it